main.go

```diff
@@ -327,6 +375,49 @@ func (z *Config) Run(params runParams) error {
 	return nil
 }
 
+var (
+    archIsRunning   bool
+    archIsRunningMu sync.Mutex
+
+)
+
+func doRun(zat *Config, params runParams){
```

Comment:
Took this out to a different method to be able to call from the server handler too.